src/relay/pass/type_infer.cc

```diff
@@ -141,17 +141,7 @@ class TypeInferencer : private ExprFunctor<Type(const Expr&)>,
   Type Unify(const Type& t1, const Type& t2, const NodeRef& expr) {
     // TODO(tqchen, jroesch): propagate span to solver
     try {
-      // instantiate higher-order func types when unifying because
-      // we only allow polymorphism at the top level
-      Type first = t1;
-      Type second = t2;
-      if (auto* ft1 = t1.as<FuncTypeNode>()) {
-        first = InstantiateFuncType(ft1);
-      }
-      if (auto* ft2 = t2.as<FuncTypeNode>()) {
-        second = InstantiateFuncType(ft2);
-      }
-      return solver_.Unify(first, second, expr);
+      return solver_.Unify(t1, t2, expr);
```

Comment:
Just curious, why did you make this change? Did you incorporate that functionality in the unifier earlier?

Reply:
I need to enable let polymorphism as anf need let polymorphism to run on cps code as cps introduce polymorphic function because the answer type is polymorphic.

Reply:
Have you added unit tests for polymorphism in lets, then? It would be good to ensure that this change isn't introducing new bugs that way.

Reply:
Yes, the test is in test_cps. now it implicitly anf cps'd code, which introduce let polymorphism.

Reply:
I meant specific unit tests for these lets, but maybe those tests are sufficient. It would be good to have such unit tests (in test_type_infer) in case something breaks with that later and we want to pinpoint what's wrong

Reply:
test_let_polymorphism